main.js

```diff
@@ -76,21 +77,76 @@ function login() {
      * @todo 1. Write code for form validation.
      * @todo 2. Fetch the auth token from backend and login the user.
      */
-}
+    const username = document.getElementById('inputUsername').value.trim();
+    const password = document.getElementById('inputPassword').value;
+
+    if (username != '' || password != '')  {
+        displayInfoToast("Please wait...");
+
+        const dataForApiRequest = {
+            username: username,
+            password: password
+        }
+
+        $.ajax({
+            url: API_BASE_URL + 'auth/login/',
+            method: 'POST',
+            data: dataForApiRequest,
+            success: function(data, status, xhr) {
+                localStorage.setItem('token', data.token);
+                window.location.href = '/';
+            },
+            error: function(xhr, status, err) {
+                displayErrorToast('Invalid username and password');
+            }
+        })
+    }
 
+}
+let index=2;
 function addTask() {
     /**
      * @todo Complete this function.
      * @todo 1. Send the request to add the task to the backend server.
      * @todo 2. Add the task in the dom.
+     * 
      */
+    const task = document.getElementById('inputTask').value.trim();
+
+    if (task != '')  {
+        displayInfoToast("Please wait...");
+
+        const dataForApiRequest = {
+            title: task
+        }
+
+        $.ajax({
+            headers: {
+                Authorization: 'Token ' + localStorage.getItem('token'),
+            },
+            
+            url: API_BASE_URL + 'todo/create/',
+            method: 'POST',
+            data: dataForApiRequest,
+            success: function(){
+                document.getElementById('inputTask').value='';
+                addSingleTask();
+                displaySuccessToast('Task Added Successfully');
+            },
+            error: function(xhr, status, err) {
+                displayErrorToast('Invalid username and password');
```

Comment:
Username and Password has no direct relation with task addition, the message should be something like - "Unable to add task, please try again."